### PR TITLE
Dispatch `iloc` calls to `getitem`

### PR DESCRIPTION
**Pull request by gforsyth**
- [x] Tests added / passed
- [x] Passes `black dask` / `flake8 dask`

Alternate solution to speeding up certain `iloc` calls xref #6345 #6264 

This is a much smaller change and also provides a nice speedup for my simple local benchmarks:
timeseries parquet file created with dask.dataframe.demo with 4 columns and 34447680 rows

current master:

```
In [3]: %time len(ddf)
CPU times: user 11.8 s, sys: 1.89 s, total: 13.7 s
Wall time: 7.1 s
Out[3]: 34447680

In [4]: %time len(ddf)
CPU times: user 10.1 s, sys: 1.6 s, total: 11.7 s
Wall time: 5.2 s
Out[4]: 34447680
```

this branch:

```
In [3]: %time len(ddf)
CPU times: user 5.9 s, sys: 777 ms, total: 6.68 s
Wall time: 3.79 s
Out[3]: 34447680

In [4]: %time len(ddf)
CPU times: user 4.01 s, sys: 504 ms, total: 4.51 s
Wall time: 1.72 s
Out[4]: 34447680
```

### CODE REVIEW

File: dask/dataframe/indexing.py [hunk 1]
```diff
@@ -58,7 +58,16 @@ def __getitem__(self, key):
...
+        if len(set(self.obj.columns)) < len(self.obj.columns):
+            # if there are any duplicate column names, do an iloc
+            return self._iloc(iindexer, cindexer)
+        else:
+            # otherwise dispatch to dask.dataframe.core.DataFrame.__getitem__
+            return self._getitem(cindexer)
+
+    def _getitem(self, cindexer):
+        col_names = self.obj.columns[cindexer]
+        return self.obj.__getitem__(col_names)
```
Comment (mrocklin) [inline dask/dataframe/indexing.py:70]: Thoughts on inlining this method?  If it's not going to be used elsewhere I wouldn't mind just sticking the two lines directly into the method above.

**Comment by mrocklin** [comment]
This PR may win the compute-hours-saved / lines-of-code contest

File: dask/dataframe/indexing.py (hunk 1 above)
Comment (gforsyth) [reply dask/dataframe/indexing.py:70]: happy to inline it, and it's unlikely to be used elsewhere.

File: dask/dataframe/indexing.py [hunk 2]
```diff
@@ -58,7 +58,13 @@ def __getitem__(self, key):
         if iindexer != slice(None):
             raise NotImplementedError(msg)
 
-        return self._iloc(iindexer, cindexer)
+        if len(set(self.obj.columns)) < len(self.obj.columns):
```
Comment (TomAugspurger) [inline dask/dataframe/indexing.py:61]:
```suggestion
        if not self.obj.columns.is_unique:
```

Probably faster than converting to sets, and is cached by pandas.

**Comment by gforsyth** [comment]
> Can you add a couple sanity tests that the result of a `df.iloc[:, ...]` is correct with duplicate columns?
> 
> * `df.iloc[:, 0]` with duplicate columns and different values for the columns with the same name
> * `df.iloc[:, list_of_integers]` selecting some duplicates and some non-duplicate columns
> * `df.iloc[:, negative_slice]`

Added these tests with duplicate columns and without.

**Comment by TomAugspurger** [comment]
Thanks Gil

**Pull request merged**